lib/hooks.php

```diff
@@ -112,4 +128,15 @@ public function onChangeUser(IUser $user, $feature, $value)
 			$this->onCreateUser($user, '');
 		}
 	}
+
+	public function onAddUserToGroup(IGroup $group, IUser $user)
+	{
+		$this->rosterPush->createOrUpdateRosterItem($user);
+	}
+
+	public function onRemoveUserFromGroup(IGroup $group, IUser $user)
+	{
+		$this->rosterPush->removeRosterItemForUsersInGroup($group, $user->getUID());
```

Comment:
TODO this must only be done when one of the privacy options is enabled and when running on version > 12.0.3